### PR TITLE
Call a script recursively with extra parameters

### PR DESCRIPTION
**Pull request by Deuchnord**
This PR adds a way to call Composer scripts with params, as described in #7562. Thus, it allows to do this:

```json
{
    "scripts": {
        "test": "phpunit",
        "testVerbose": "@test -vvv"
    }
}
```

Calling `testVerbose` will resolve as `phpunit -vvv`.

This PR also adds some docs about this new feature.

### CODE REVIEW

**Comment by alcohol** [comment]
Would be great if you could add a testcase.

File: doc/articles/scripts.md [hunk 1]
```diff
@@ -238,6 +238,17 @@ one by prefixing the command name with `@`:
 }
 ```
 
+You can also refer a script and pass it new arguments:
+
+```json
+{
+  "scripts": {
+    "tests": "phpunit",
+    "testsVerbose": "@unitTests -vvv"
```
Comment (staabm) [inline doc/articles/scripts.md]: Looks like a typo?

```suggestion
    "testsVerbose": "@tests -vvv"
```

Comment (alcohol) [reply doc/articles/scripts.md]: Indeed.

Comment (Deuchnord) [reply doc/articles/scripts.md]: Thanks :)

Comment (staabm) [reply doc/articles/scripts.md]: this new github „suggestd change“ feature is sooo cool :)

Comment (Deuchnord) [reply doc/articles/scripts.md]: I agree, just discovered, that's really a good idea :D

**Comment by Deuchnord** [comment]
> Would be great if you could add a testcase.

OK, I'm going to do it a little later :)

**Comment by Deuchnord** [comment]
Just added unit tests, sorry for the late :)

**Comment by Deuchnord** [comment]
AppVeyor seems having had an internal issue, should be restarted

**Comment by alcohol** [comment]
Yeah I tried to do that but it seems I cannot :-(

**Comment by Deuchnord** [comment]
I've just amended my last commit and forced a push to force the checks restarting ;)

**Comment by Deuchnord** [comment]
Returned by AppVeyor:
```
There was 1 failure:
1) Composer\Test\EventDispatcher\EventDispatcherTest::testRecursionInScriptsNames
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
 '> helloWorld: @hello World
-> hello: echo Hello 'World'
+> hello: echo Hello "World"
 '
C:\projects\composer\tests\Composer\Test\EventDispatcher\EventDispatcherTest.php:314
```

I don't have Windows, so it will be a little tricky for me to find what's happening here… Can anybody guide me for that quotes issue? :confused:

**Comment by sbuzonas** [comment]
@Deuchnord [escapeshellarg](http://php.net/manual/en/function.escapeshellarg.php#74815) apparently does double quotes on windows

**Comment by Deuchnord** [comment]
Thanks for this hint @sbuzonas!

**Comment by Seldaek** [comment]
Thanks

**Pull request merged**